### PR TITLE
Hand pose

### PR DESCRIPTION
**Pull request by vinicius121**
Sample to detect the hand key points with the OpenPose model.

### CODE REVIEW

File: SamplesCS/Samples/HandPose.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                points.Add(maxLoc);
+            }
+
+            int nPairs = 20;//(POSE_PAIRS).Length / POSE_PAIRS[0].Length;
+
+            for (int n = 0; n < nPairs; n++)
+            {
+                // lookup 2 connected body/hand parts
+                Point2f partA = points[POSE_PAIRS[n][0]];
+                Point2f partB = points[POSE_PAIRS[n][1]];
```
Comment (shimat) [inline SamplesCS/Samples/HandPose.cs]:
```suggestion
                Point partB = points[POSE_PAIRS[n][1]];
```

File: SamplesCS/Samples/HandPose.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+                points.Add(maxLoc);
+            }
+
+            int nPairs = 20;//(POSE_PAIRS).Length / POSE_PAIRS[0].Length;
+
+            for (int n = 0; n < nPairs; n++)
+            {
+                // lookup 2 connected body/hand parts
+                Point2f partA = points[POSE_PAIRS[n][0]];
```
Comment (shimat) [inline SamplesCS/Samples/HandPose.cs]:
```suggestion
                Point partA = points[POSE_PAIRS[n][0]];
```

File: SamplesCS/Samples/HandPose.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                Point2f partB = points[POSE_PAIRS[n][1]];
+
+                if (partA.X <= 0 || partA.Y <= 0 || partB.X <= 0 || partB.Y <= 0)
+                    continue;
+
+                Cv2.Line(frame, (OpenCvSharp.Point)partA, (OpenCvSharp.Point)partB, new Scalar(0, 255, 255), 8);
+                Cv2.Circle(frame, (int)partA.X, (int)partA.Y, 8, new Scalar(0, 0, 255), -1);
+                Cv2.Circle(frame, (int)partB.X, (int)partB.Y, 8, new Scalar(0, 0, 255), -1);
```
Comment (shimat) [inline SamplesCS/Samples/HandPose.cs]:
```suggestion
                Cv2.Circle(frame, partB.X, partB.Y, 8, new Scalar(0, 0, 255), -1);
```

File: SamplesCS/Samples/HandPose.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            {
+                // lookup 2 connected body/hand parts
+                Point2f partA = points[POSE_PAIRS[n][0]];
+                Point2f partB = points[POSE_PAIRS[n][1]];
+
+                if (partA.X <= 0 || partA.Y <= 0 || partB.X <= 0 || partB.Y <= 0)
+                    continue;
+
+                Cv2.Line(frame, (OpenCvSharp.Point)partA, (OpenCvSharp.Point)partB, new Scalar(0, 255, 255), 8);
+                Cv2.Circle(frame, (int)partA.X, (int)partA.Y, 8, new Scalar(0, 0, 255), -1);
```
Comment (shimat) [inline SamplesCS/Samples/HandPose.cs]:
```suggestion
                Cv2.Circle(frame, partA.X, partA.Y, 8, new Scalar(0, 0, 255), -1);
```

File: SamplesCS/Samples/HandPose.cs [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            for (int n = 0; n < nPairs; n++)
+            {
+                // lookup 2 connected body/hand parts
+                Point2f partA = points[POSE_PAIRS[n][0]];
+                Point2f partB = points[POSE_PAIRS[n][1]];
+
+                if (partA.X <= 0 || partA.Y <= 0 || partB.X <= 0 || partB.Y <= 0)
+                    continue;
+
+                Cv2.Line(frame, (OpenCvSharp.Point)partA, (OpenCvSharp.Point)partB, new Scalar(0, 255, 255), 8);
```
Comment (shimat) [inline SamplesCS/Samples/HandPose.cs]:
```suggestion
                Cv2.Line(frame, partA, partB, new Scalar(0, 255, 255), 8);
```

**Review by shimat — COMMENTED** [review]
Thanks!

File: SamplesCS/Samples/FaceDetectionDNN.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    /// Add the files to the bin folder
+    /// </summary>
+    class FaceDetectionDNN : ISample
+    {
+        public void Run()
+        {
+            string finalOutput = "DetectedFaces.jpg";
+            string faceModel = "res10_300x300_ssd_iter_140000_fp16.caffemodel";
+			string configFile = "deploy.prototxt";
+			var image = "faces.jpg";
```
Comment (shimat) [inline SamplesCS/Samples/FaceDetectionDNN.cs]: replace tabs :pray:

Comment (vinicius121) [reply SamplesCS/Samples/FaceDetectionDNN.cs]: I think it is also done now!

**Pull request merged**